### PR TITLE
Response validation middleware

### PR DESCRIPTION
**Pull request by sondrelg**

### CODE REVIEW

**Review by sondrelg — COMMENTED** [review]
Missing docs and wrapper

**Comment by sondrelg** [comment]
Ready to review - ill add docs in a separate PR

**Review by sondrelg — COMMENTED** [review]
Looks OK

File: tests/test_middleware/test_response_handling.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+def test_response_containing_uuid(client, caplog):
```
Comment (JonasKs) [inline tests/test_middleware/test_response_handling.py:1]: Caplog not needed.

File: django_swagger_tester/utils.py [hunk 2]
```diff
@@ -255,3 +296,113 @@ def type_placeholder_value(_type: str) -> Any:
...
+    """
+    Loads response data as JSON and returns a copied response object.
+    """
+    # By parsing the response data JSON we bypass problems like uuid's not having been converted to
+    # strings yet, which otherwise would create problems when comparing response data types to the
+    # documented schema types in the schema tester
+    content = response.content.decode(response.charset)
+    response_data = json.loads(content)
+    copied_response = deepcopy(response)
+    copied_response.data = response_data  # this can probably be done differently
```
Comment (JonasKs) [inline django_swagger_tester/utils.py:375]: Not sure why it needs to be done differently? This looks clean to me. 
I’d remove the comment and just return it here.

**Review by JonasKs — APPROVED** [review]
Looks super solid to me. I’ve only been able to look through on my phone, but can’t see anything obvious to comment on or that I’d want to test out. 

You’re a beast.

**Pull request merged**